homeassistant/helpers/aiohttp_client.py

```diff
@@ -93,6 +90,7 @@ def async_get_clientsession(
 def async_create_clientsession(
     hass: HomeAssistant,
     verify_ssl: bool = True,
+    force_close: bool = False,
```

Comment:
let's not expand our helper. Instead, let's just keep this logic inside Octoprint. Or even better, shouldn't `pyoctoprintapi` just stop accepting a client session if it's requiring special configuration ?

Reply:
That would also be an option, yes. I am not sure where the best place for this would be, but I think naively it makes sense to _not_ put the session into the library, since it is meant to be reused on the application side. However, in this case I also think that this "quirk" of flask/OctoPrint might warrant putting a prepared session into the library itself.

If you think it is not ok to do this fix within homeassistant itself, I can create a PR on the library side instead (and hope it is still maintained 😅).

I also think that it might be possible to fix this from the OctoPrint Webserver side (which is using flask), however looking into this didn't bring up any feasible option for me. If this actually _is_ possible one of the OctoPrint maintainers would have to take a deeper look.

Reply:
We also have a custom aiohttp session for aiohue as we ran into a similar problem.

Reply:
Oh I think I misunderstood, you really only meant to not expand the helper, but still put the custom session into the octoprint package within home assistant, right?

That would obviously also be an option, however I would not be sure what options to include in that case. It seems like Home Assistant uses a lot of things when the helper is used. When the session is provided by the lib I would not care, but when adapting the integration might be of interest?

Reply:
the defaults of aiohttp are ok. 

Nope, I didn't mean in the integration but in actual `pyoctoprintapi`. This is what aiohue does: https://github.com/home-assistant-libs/aiohue/blob/master/aiohue/v2/__init__.py#L196-L200

Reply:
Hmm yeah idk, for me both putting it in the integration as well as putting it into `pyoctoprintapi` seem like equally good solutions, if there is no easy way to fix this in OctoPrint itself. However, since I am working in a project that isn't mine (neither of the three) I cannot guarantee for anything. Also because of this, I would kindly ask for you to decide in which way I should proceed with this. 

Can I implement the session in hass for now?
Should I open a PR in `pyoctoprintapi` and integrate that later into hass?
Or should I wait for OctoPrint to fix it on the server side?

I am happy to do any of them, and if there is a better way down the line I can come back to this and update the implementation. But even if there is I feel like doing it manually inside of Home Assistant (like auihue does it) would be the fastest option for now, since it would allow the integration to work immediately, even if it is changed lateron.

Reply:
Please make a PR to `pyoctoprintapi`. If OctoPrint adjust their server the custom session can be removed at that point.